aten/src/ATen/native/Distributions.h

```diff
@@ -166,7 +166,13 @@ C10_DEVICE scalar_t binomial_inversion(scalar_t count, scalar_t prob, BaseSample
 
   while (1) {
     U = standard_uniform.sample();
-    accscalar_t geom = compat_ceil(compat_log(U) / logprob);
+    // Cuda's curand_uniform can return 1.0 but not 0.0
+    // Reject 1.0 since it will produce geom == 0.0
+    accscalar_t logU = compat_log(U);
+    if (logU == 0.0) {
```

Comment:
Done. I was hesitant to do such a sweeping change at first in fear it might change values for people who use a seed and expect consistent results. Is there a stream-constant guarantee in PyTorch?

Reply:
There are no guarantees that underlying generation techniques would not change, e.g. when the whole algorithm for generating Binomial changed it was ok.